src/meta/partition_guardian.cpp

```diff
@@ -658,7 +658,8 @@ void partition_guardian::finish_cure_proposal(meta_view &view,
             np->newly_remove_primary(gpid.get_app_id(), false);
         } else if (act.type == config_type::CT_UPGRADE_TO_PRIMARY) {
             np->newly_remove_primary(gpid.get_app_id(), true);
-        } else if (act.type == config_type::CT_UPGRADE_TO_SECONDARY) {
+        } else if (act.type == config_type::CT_UPGRADE_TO_SECONDARY ||
+                   act.type == config_type::CT_ADD_SECONDARY) {
```

Comment:
```suggestion
        } else if (act.type == config_type::CT_UPGRADE_TO_SECONDARY){
            np->newly_remove_partition(gpid.get_app_id());
        } else if (act.type == config_type::CT_ADD_SECONDARY){
            np->newly_remove_partition(gpid.get_app_id());
        }
```
It's more clearly

Reply:
I don't think so. It doesn't make sense

Reply:
ref:
https://github.com/XiaoMi/rdsn/blob/3dedd1e98816d639821ccc9b0139976230bc670c/src/meta/partition_guardian.cpp#L657-L663

you can refactor it using `switch` or keep the original style.

Reply:
okay